tests/main/nss-modules/task.yaml

```diff
@@ -0,0 +1,315 @@
+summary: Run a snap as an LDAP user
+
+details: |
+    Test that a user defined in LDAP can run a snap without errors or warnings.
+    The system setup is mostly taken from the sssd tests found at 
+    https://git.launchpad.net/ubuntu/+source/sssd/tree/debian/tests
+
+systems: 
+    - -fedora-*
+    - -debian-*
+    - -arch-*
+    - -opensuse-*
+    - -centos-9-*
+    - -amazon-*
+    - -ubuntu-core*
+    - -ubuntu-16.04-64
+
+environment:
+    domain_name: example.com
+    host_name: server.${domain_name}
+    ldap_suffix: dc=example,dc=com
+    admin_dn: cn=admin,${ldap_suffix}
+    admin_pw: secret
+    nss_user: nsstestuser1
+    nss_user_pw: testuser1secret
+    ldap_group: ldapusers
+    realm_name: EXAMPLE.COM
+    kerberos_principal_pw: testuser1kerberos
+    MODULE/ldap: ldap
+    MODULE/sss: sss
+    MODULE/nis: nis
+    MODULE/winbind: winbind
+
+prepare: |
+    if [[ "$SPREAD_BACKEND" =~ openstack ]] && os.query is_arm && { [[ "$MODULE" = "nis" ]] || [[ "$MODULE" = "winbind" ]]; }; then
+        tests.exec skip-test "The openstack ARM environment is not KVM-enabled and cannot run LXD VMs" && exit 0
+    fi
+    backup_if_exists() {
+        local item=$1
+        if [ -f "$item" ]; then
+            cp "$item" "$item.bak"
```

Comment:
Maybe it would be nice to fail if `"$item.bak"` exists.

Reply:
Could we use the back-up helper and extend it to handle this gracefully, if required?